src/services/authService.ts

```diff
@@ -0,0 +1,136 @@
+import { verify, sign } from 'hono/jwt';
+import { authRepository } from '../dataAcces/authRepository';
+import { userRepository } from '../dataAcces/userRepository';
+import { DbError, ValidationError } from '../errors/errors';
+import bcrypt from 'bcrypt';
+import { type User } from '@prisma/client';
+import { userService } from './userService';
+
+export const authService = {
+  async generateTokens({ id, email }: Record<string, unknown>) {
+    if (!process.env.JWT_ACCESS_SECRET || !process.env.JWT_REFRESH_SECRET) {
+      throw Error(
+        'You need JWT_ACCESS_SECRET and JWT_REFRESH_SECRET environmental variables.'
+      );
+    }
+
+    const accessToken = await sign(
+      { id, email },
+      process.env.JWT_ACCESS_SECRET
+    ); // expired?
+    const refreshToken = await sign(
+      { id, email },
+      process.env.JWT_REFRESH_SECRET
+    );
+
+    return { accessToken, refreshToken };
+  },
+
+  async saveToken(userId: number, refreshToken: string) {
+    const tokenData = await authRepository.getByUserId(userId);
+
+    if (tokenData) {
+      tokenData.refreshToken = refreshToken;
+      return await authRepository.update(tokenData);
+    }
+
+    const token = await authRepository.create({ userId, refreshToken });
+
+    return token;
+  },
+
+  async signUp(user: User) {
+    const { login, email, password } = user;
+    const newUser = await userService.create({
+      login,
+      email,
+      password,
+      role: 'USER',
+    });
+
+    const tokens = await this.generateTokens({ id: newUser.id, email });
```

Comment:
Not sure if we need to log in the user immediately after the registration. Let’s separate these logics for now